lib/lte_link_control/lte_lc.c

```diff
@@ -1082,56 +1083,49 @@ int lte_lc_edrx_req(bool enable)
 		}
 	}
 
-	switch (sys_mode_current) {
-	case LTE_LC_SYSTEM_MODE_LTEM:
-	case LTE_LC_SYSTEM_MODE_LTEM_GPS:
-		actt = AT_CEDRXS_ACTT_WB;
-		break;
-	case LTE_LC_SYSTEM_MODE_NBIOT:
-	case LTE_LC_SYSTEM_MODE_NBIOT_GPS:
-		actt = AT_CEDRXS_ACTT_NB;
-		break;
-	default:
-		LOG_ERR("Cannot request eDRX for this system mode (%d)",
-			sys_mode_current);
-		return -EOPNOTSUPP;
-	}
-
-	if (enable) {
-		if (strlen(edrx_param) == 4) {
-			snprintf(req, sizeof(req),
-				"AT+CEDRXS=2,%d,\"%s\"", actt, edrx_param);
+	for (size_t i = 0; i < ARRAY_SIZE(actt); i++) {
+		if (enable) {
+			if (strlen(edrx_param) == 4) {
```

Comment:
Does this check ever fail? Looks like ```edrx_param``` is strictly enforced to 4 characters by the library. And if the check fails, shouldn't the API give out and error because the max length of the eDRX param has been violated.

Reply:
The length can also be zero, set by Kconfig, so the check is needed, at least with the current implemenation